File: leetcode-questions/csharp/src/test/csharp/com/leetcode/questions/validparentheses/ValidParenthesesTest.cs
using Xunit;
using System.Collections.Generic;
using com.leetcode.questions.leetcode.editor.en.validparentheses;
using com.leetcode.questions.support;

namespace com.leetcode.questions.validparentheses;

public class ValidParenthesesTest {

    private const string CasesFileName = "valid-parentheses.json";

    public ValidParenthesesTest() {

    }

    [Fact]
    public void TestCases() {
        GeneratedLeetCodeTestHarness.Run(typeof(Solution), "IsValid", GeneratedLeetCodeTestHarness.ReadCasesFile(CasesFileName));
    }
}

File: leetcode-questions/java/src/test/java/com/leetcode/questions/validparentheses/ValidParenthesesTest.java
package com.leetcode.questions.validparentheses;

import com.leetcode.questions.leetcode.editor.en.ValidParentheses;
import com.leetcode.questions.support.GeneratedLeetCodeTestHarness;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.nio.file.Files;
import java.nio.file.Path;

public class ValidParenthesesTest {

    private static final Path CASES_PATH = Path.of("generated-test-cases", "valid-parentheses.json");

    public ValidParenthesesTest() {

    }

    @Test
    public void testCases() throws Exception {
        GeneratedLeetCodeTestHarness.run(ValidParentheses.Solution.class, "isValid", Files.readString(CASES_PATH));
    }
}
